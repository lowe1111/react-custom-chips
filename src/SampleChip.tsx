import React, { FC } from 'react';
import styled from '@emotion/styled';
import { keyframes } from '@emotion/core';

import { RemovableChipData } from './chip.interface';

const growChip = keyframes`
  0% {
    transform: scale(0)
  }
  60% {
    transform: scale(1.1)
  }
  100% {
    transform: scale(1)
  }
`;

const Chip = styled.div`
  padding: 5px 10px 5px 5px;
  margin: 5px;
  text-transform: uppercase;
  display: flex;
  border-radius: 50px;
  background-color: lightgrey;
  font-size: 12px;
  font-weight: 600;
  line-height: 2.17;
  letter-spacing: 0.2px;
  align-items: center;
  color: darkgrey;
  animation: ${growChip} 250ms cubic-bezier(0.32, 0.62, 0.2, 0.88);
  position: relative;
  cursor: pointer;
  &:after {
    pointer-events: none;
    content: '';
    position: absolute;
    width: 100%;
    height: 100%;
    border-radius: 50px;
    background-color: black;
    opacity: 0.1;
    margin-left: -5px;
    display: none;
  }
  > span {
    margin-left: 10px;
  }
  &:hover {
    &:after {
      display: block;
    }
  }
`;

interface Props {
  value: RemovableChipData;
}

const SampleChip: FC<Props> = ({ value }) => {
  const removeThisChip = () => value.onRemove(value);
  return (
    <Chip>{value.name} <span onClick={removeThisChip}>X</span></Chip>
  );
}

export default SampleChip;
