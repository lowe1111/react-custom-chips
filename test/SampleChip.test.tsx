import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import SampleChip from '../src/SampleChip';

describe('SampleChip', () => {
  beforeAll(() => {
    Enzyme.configure({ adapter: new Adapter() });
  });

  test('renders correctly', () => {
    const onRemoveFunc = jest.fn();
    const component = shallow(
      <SampleChip
        value={{ id: '1', name: 'chip', onRemove: onRemoveFunc}}
      />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });
  test('click close triggers onRemove', () => {
    const onRemoveFunc = jest.fn();
    const component = shallow(
      <SampleChip
        value={{ id: '1', name: 'chip', onRemove: onRemoveFunc}}
      />,
    );
    component.find('.remove').simulate('click');
    expect(onRemoveFunc).toBeCalled();
    expect(toJson(component)).toMatchSnapshot();
  });
});
