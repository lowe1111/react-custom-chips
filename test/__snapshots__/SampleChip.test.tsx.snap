// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SampleChip click close triggers onRemove 1`] = `
<div
  className="chip"
>
  chip
  <div
    className="remove"
    onClick={[Function]}
  >
    <CloseIcon />
  </div>
</div>
`;

exports[`SampleChip renders correctly 1`] = `
<div
  className="chip"
>
  chip
  <div
    className="remove"
    onClick={[Function]}
  >
    <CloseIcon />
  </div>
</div>
`;
